### PR TITLE
Join case bug and enhancement

### PR DESCRIPTION
**Pull request by nitish-beehyv**
Join case bug and enhacement

### CODE REVIEW

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js [hunk 1]
```diff
@@ -5,17 +5,18 @@ import {
...
 } from "@egovernments/digit-ui-react-components";
 import { useHistory } from "react-router-dom";
-import React, { useEffect, useState } from "react";
-import { Link } from "react-router-dom";
+import React, { useEffect, useRef, useState } from "react";
 import { InfoCard } from "@egovernments/digit-ui-components";
 import { DRISTIService } from "../../../../dristi/src/services";
 import { RightArrow } from "../../../../dristi/src/icons/svgIndex";
 import { CASEService } from "../../hooks/services";
+import { isEqual } from "lodash";
```
Comment (anirudh-0) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js:19]: replace with `import isEqual from "lodash/isEqual";`

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js [hunk 2]
```diff
@@ -66,19 +67,124 @@ const JoinHomeLocalisation = {
   CONFIRM_ATTENDANCE: "CONFIRM_ATTENDANCE",
   JOINING_THIS_CASE_AS: "JOINING_THIS_CASE_AS",
   SKIP_LATER: "SKIP_LATER",
+  CASE_NO_ADMITTED_STATUS: "The above case doesn't have admitted status",
```
Comment (anirudh-0) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js:70]: localization should be there for this line

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js [hunk 3]
```diff
@@ -145,11 +237,44 @@ const JoinCaseHome = ({ t }) => {
...
   };
 
+  const onFormValueChange = (formData) => {
+    // const temp = formData?.advocateBarRegNumberWithName?.[0]?.data;
+
+    if (!isEqual(formData, advocateDetailForm)) {
+      setAdvocateDetailForm(formData);
+    }
+
+    // setAdvocateDetail({
```
Comment (anirudh-0) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js:247]: remove unnecessary commented code

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js [hunk 4]
```diff
@@ -145,11 +237,44 @@ const JoinCaseHome = ({ t }) => {
...
+    // }
+    setIsDisabled(false);
+  };
+
   useEffect(() => {
-    // console.log('step', step)
     if (step === 0) {
-      if (caseNumber !== "") setIsDisabled(false);
-      else setIsDisabled(true);
+      // if (caseNumber !== "" && caseNumber.length > 1) {
```
Comment (anirudh-0) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js:270]: step 0 has no operations ???

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js [hunk 5]
```diff
@@ -390,19 +639,35 @@ const JoinCaseHome = ({ t }) => {
               <InfoCard
                 variant={"default"}
                 label={userType === "Litigant" ? t(JoinHomeLocalisation.PLEASE_NOTE) : t("INFO")}
-                additionalElements={{}}
+                additionalElements={
```
Comment (anirudh-0) [inline frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js:642]: check file for all localization misses

**Review by anirudh-0 — CHANGES_REQUESTED** [review]

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js (hunk 1 above)
Comment (anirudh-0) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js:19]: code isn't even compiling

Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/cases/src/pages/employee/JoinCaseHome.js]: I made the changes.

**Pull request merged**